lib/classes/ConfigSchemaHandler.test.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+const expect = require('chai').expect;
+const ConfigSchemaHandler = require('./ConfigSchemaHandler');
+const sinon = require('sinon');
+const Serverless = require('../Serverless');
+
+describe('ConfigSchemaHandler', () => {
+  describe('#constructor', () => {
+    it('should have schema property', () => {
+      const configSchemaHandler = new ConfigSchemaHandler();
```

Comment:
Here we also should test with `runServeless` (and no need to test internal characteristics which is considered private)

Reply:
Sorry, I don't see how `runServerless` should be applied in this file. Initially, I copied similar `it` test from `lib/configSchema/index.js` where I added`runServerless` but then I realized there is nothing to change here. So I don't understand why duplicating the same logic... 

Currently, the test described in `#validateConfig` ensures that `cli.log` is called (on not called) in case of failing (or passing) validation. 

If you still think that a test with `runServerless` should be here, could you be please be more specific about what exactly should this test ensure?

> (and no need to test internal characteristics which is considered private)

Could you please explain this in more detail? I'm not sure I understand this

Reply:
I think we should test that functions that extend schemas works.

So create a fixture that introduces:
- A dummy provider plugin, which configures some event
- `serverless.yml` config that configures a function for given provider and event.

(I've outlined how it can be done here: https://github.com/serverless/serverless/pull/7335#discussion_r389322881)

Then in `runServerless` use that fixture and confirm that:
- Given `serveless.yml` is validated without issues (technically successful run of `runServeless` does that)
- Confirm that schema is extended as expected by plugin

Reply:
All clear, thanks.  I've made this refactoring.

One thing I noticed though. For some reason, `runServerless` loads all AWS events, even though `awsProvider` plugin is not loaded. (To give more context, I noticed it in newly created `configSchemaExtensions` fixture which does not load it). However, I assume this is not in the scope of this PR.

Reply:
Yes, it's a design flaw which could be improved (but yes, it's totally not in scope here)

You just need to confirm in each plugin that we deal with AWS provider service and extend schema only in such case.

Reply:
Sure thing. Done in https://github.com/serverless/serverless/pull/7335/commits/11140beb8f2b8935540630e751684f2bcb81f34d